src/main/java/duke/dukehelper/CommandHandler.java

```diff
@@ -0,0 +1,206 @@
+            throw new DukeException.NoList();
+        }
+    }
+    /**
+     * Handles command and prints messages, if any onto console depending on input parameter.
+     *
+     * @param command a string of a simple root word to represent a particular message
+     * @param packet an input parameter for the message to print, given the command requires further inputs.
+     */
+    public static void handleCommand(Constants.Command command, Packet packet){
+        String output = "";
+        String customErrorMessage = "";
+        boolean isDrawPartition = true;
+        Constants.Error err = null;
+
+        switch(command){
+        case HELLO:
+            output = UiManager.MESSAGE_LOGO;
+            output += UiManager.MESSAGE_HELLO;
+            break;
+
+        case BYE:
+            output = UiManager.MESSAGE_BYE;
+            break;
+
+        case ECHO:
+            output = String.format("%s\n", packet.getPacketPayload());
+            break;
+
+        case PROMPT_INPUT:
+            output = ">>> ";
+            isDrawPartition = false;
+            break;
+
+        case INSERT_TASK_DEADLINE:
+            //Fall through
+        case INSERT_TASK_TODO:
+            //Fall through
+        case INSERT_TASK_EVENT:
+            /**
+             *  The exception handling task is split into 2 types.
+             *  The first exception type should be detected by the command handler; If the command needs a
+             *  description and the input does not supply any, we throw an exception here.
+             *  The second exception type is unique to what the different variants of tasks want,
+             *  so we encapsulate the task-unique exception handling within their own class.
+             */
+            label: try {
+                validatePayload(packet);
+                Task inputEvent = generateTask(command, packet);
+                err = list.addTask(inputEvent);
+                if (err != Constants.Error.NO_ERROR) {
+                    break label;
+                }
+                output = UiManager.getMessageTaskAdded(inputEvent);
+                output += UiManager.getMessageReportNumTasks(list);
+            } catch (DukeException.InvalidDescription exception) {
+                customErrorMessage = "You have not entered a valid description!\n";
+                DukeException.printErrorMessage(Constants.Error.WRONG_ARGUMENTS, customErrorMessage);
+            } finally {
+                if (err != Constants.Error.NO_ERROR ) {
+                    DukeException.printErrorMessage(Constants.Error.TASK_NOT_CREATED);
+                }
+            }
+            break;
+
+        case SHOW_LIST:
+            if (list.getNumTasks() == 0){
+                DukeException.printErrorMessage(Constants.Error.NO_LIST);
+            } else {
+                output = list.showAllTasks();
+            }
+            break;
+
+        case REMOVE_TASK:
+            // Fall through
+        case MARK_TASK_DONE:
```

Comment:
Great job making sure that all possible scenarios are dealt with. I'm not sure if I missed it, but perhaps you can consider the case of trying to do a task that is already done? It's not exactly an error but it would be great if your code recognises that and acknowledges it too.

Reply:
Great suggestion! I will implement it. Thanks!